api/v1/inferencepool_types.go

```diff
@@ -96,26 +100,26 @@ type Extension struct {
 	//
 	// +optional
 	// +kubebuilder:default=Service
-	Kind *Kind `json:"kind,omitempty"`
+	Kind Kind `json:"kind,omitempty"`
 
 	// Name is the name of the referent.
 	//
-	// +kubebuilder:validation:Required
-	Name ObjectName `json:"name"`
+	// +required
+	Name ObjectName `json:"name,omitempty"`
 
 	// The port number on the service running the extension. When unspecified,
 	// implementations SHOULD infer a default value of 9002 when the Kind is
 	// Service.
 	//
 	// +optional
-	PortNumber *PortNumber `json:"portNumber,omitempty"`
+	PortNumber PortNumber `json:"portNumber,omitempty"`
 
 	// Configures how the gateway handles the case when the extension is not responsive.
 	// Defaults to failClose.
 	//
 	// +optional
 	// +kubebuilder:default="FailClose"
-	FailureMode *ExtensionFailureMode `json:"failureMode"`
+	FailureMode ExtensionFailureMode `json:"failureMode,omitempty"`
```

Comment:
The field should be a pointer.

Reply:
@JoelSpeed I'm wondering whether it conflicts with https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#serialization-of-optionalrequired-fields here?

This field is optional and zero value "" is not a valid choice and the the field type have a built-in nil value (map or slice), seems like we need to use pointer?